testsuite/MDAnalysisTests/core/test_universe.py

```diff
@@ -707,6 +707,63 @@ def test_add_connection(self, universe, attr, values):
     def add_connection_error(self, universe, attr, values):
         with pytest.raises(ValueError):
             universe.add_TopologyAttr(attr, values)
+
+
+class TestDelTopologyAttr(object):
+    @pytest.fixture()
+    def universe(self):
+        u = make_Universe(("masses", "charges", "resnames"))
+        u.add_TopologyAttr("bonds", [(0, 1)])
+        return u
+
+    def test_del_TA_fail(self, universe):
+        with pytest.raises(ValueError, match="Unrecognised"):
+            universe.del_TopologyAttr('silly')
+
+    def test_absent_fail(self, universe):
+        with pytest.raises(ValueError, match="not in Universe"):
+            universe.del_TopologyAttr("angles")
+
+    def test_wrongtype_fail(self, universe):
+        with pytest.raises(ValueError, match="must be str or TopologyAttr"):
+            universe.del_TopologyAttr(list)
+
+    @pytest.mark.parametrize(
+        'todel,attrname', [
+            ("charge", "charges"),
+            ("charges", "charges"),
+            ("bonds", "bonds"),
+        ]
+    )
+    def test_del_str(self, universe, todel, attrname):
+        assert hasattr(universe.atoms, attrname)
+        universe.del_TopologyAttr(todel)
+        assert not hasattr(universe.atoms, attrname)
+
+    def test_del_attr(self, universe):
+        assert hasattr(universe.atoms, "resnames")
+        assert hasattr(universe.residues, "resnames")
+        assert hasattr(universe.segments, "resnames")
+        assert hasattr(universe.atoms[0], "resname")
+        assert hasattr(universe.residues[0], "resname")
+        universe.del_TopologyAttr(universe._topology.resnames)
+        assert not hasattr(universe.atoms, "resnames")
+        assert not hasattr(universe.residues, "resnames")
+        assert not hasattr(universe.segments, "resnames")
+        assert not hasattr(universe.atoms[0], "resname")
+        assert not hasattr(universe.residues[0], "resname")
+
+    def test_del_transplants(self, universe):
+        atom = universe.atoms[0]
+        assert hasattr(atom, "fragindex")
+        universe.del_TopologyAttr(universe.bonds)
+        assert not hasattr(atom, "fragindex")
+        assert not hasattr(universe.atoms, "fragindices")
+
+    def test_del_attr_noerror(self, universe):
+        # does not raise AttributeError or ValueError
```

Comment:
This should be an error though?

Reply:
Added a test that an error is raised, and preview docs [here](https://mdanalysis--3069.org.readthedocs.build/en/3069/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.del_TopologyAttr) 🥳